### PR TITLE
[incubator/druid] Use mainline druid image, flexible configuration with envvars

### LINKED ISSUE #11484: [incubator/druid] Middlemanager tries to access S3 despite of `druid.storage.type=local`
**Is this a request for help?**: No

---

**Is this a BUG REPORT or FEATURE REQUEST?** (choose one): BUG REPORT

**Version of Helm and Kubernetes**:

$ helm version
Client: &version.Version{SemVer:"v2.12.2", GitCommit:"7d2b0c73d734f6586ed222a567c5d103fed435be", GitTreeState:"clean"}
Server: &version.Version{SemVer:"v2.12.2", GitCommit:"7d2b0c73d734f6586ed222a567c5d103fed435be", GitTreeState:"clean"}

$ kubectl version
Client Version: version.Info{Major:"1", Minor:"13", GitVersion:"v1.13.0", GitCommit:"ddf47ac13c1a9483ea035a79cd7c10005ff21a6d", GitTreeState:"archive", BuildDate:"2018-12-14T20:49:34Z", GoVersion:"go1.11.3", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"13", GitVersion:"v1.13.3", GitCommit:"721bfa751924da8d1680787490c54b9179b1fed0", GitTreeState:"clean", BuildDate:"2019-02-01T20:00:57Z", GoVersion:"go1.11.5", Compiler:"gc", Platform:"linux/amd64"}

**Which chart**:

incubator/druid

Chart version: 0.1.0 (currently only version)
Using image: maver1ckpl/druid-docker:0.12.3-2

**What happened**:

I installed `incubator/druid` with default values.
I tried to ingest a kafka topic by posting a proper spec to `druid/indexer/v1/supervisor`.

The ingestion task fail and the logs in `/var/druid/indexing-logs` say:

```
2019-02-16T05:50:37,570 ERROR [main] io.druid.cli.CliPeon - Error when starting up.  Failing.
com.google.inject.ProvisionException: Unable to provision, see the following errors:

1) Error in custom provider, com.amazonaws.AmazonClientException: Unable to load AWS credentials from any provider in the chain
  at io.druid.storage.s3.S3StorageDruidModule.getRestS3Service(S3StorageDruidModule.java:107) (via modules: com.google.inject.util.Modules$OverrideModule -> io.druid.storage.s3.S3StorageDruidModule)
  at io.druid.storage.s3.S3StorageDruidModule.getRestS3Service(S3StorageDruidModule.java:107) (via modules: com.google.inject.util.Modules$OverrideModule -> io.druid.storage.s3.S3StorageDruidModule)
  while locating org.jets3t.service.impl.rest.httpclient.RestS3Service
    for the 1st parameter of io.druid.storage.s3.S3DataSegmentKiller.<init>(S3DataSegmentKiller.java:46)
  while locating io.druid.storage.s3.S3DataSegmentKiller
  at io.druid.storage.s3.S3StorageDruidModule.configure(S3StorageDruidModule.java:87) (via modules: com.google.inject.util.Modules$OverrideModule -> io.druid.storage.s3.S3StorageDruidModule)
  while locating io.druid.segment.loading.DataSegmentKiller annotated with @com.google.inject.multibindings.Element(setName=,uniqueId=148, type=MAPBINDER, keyType=java.lang.String)
  at io.druid.guice.Binders.dataSegmentKillerBinder(Binders.java:46) (via modules: com.google.inject.util.Modules$OverrideModule -> io.druid.storage.s3.S3StorageDruidModule -> com.google.inject.multibindings.MapBinder$RealMapBinder)
  while locating java.util.Map<java.lang.String, io.druid.segment.loading.DataSegmentKiller>
    for the 1st parameter of io.druid.segment.loading.OmniDataSegmentKiller.<init>(OmniDataSegmentKiller.java:39)
  while locating io.druid.segment.loading.OmniDataSegmentKiller
  at io.druid.cli.CliPeon$1.configure(CliPeon.java:176) (via modules: com.google.inject.util.Modules$OverrideModule -> com.google.inject.util.Modules$OverrideModule -> io.druid.cli.CliPeon$1)
  while locating io.druid.segment.loading.DataSegmentKiller
    for the 5th parameter of io.druid.indexing.common.TaskToolboxFactory.<init>(TaskToolboxFactory.java:108)
  at io.druid.cli.CliPeon$1.configure(CliPeon.java:165) (via modules: com.google.inject.util.Modules$OverrideModule -> com.google.inject.util.Modules$OverrideModule -> io.druid.cli.CliPeon$1)
  while locating io.druid.indexing.common.TaskToolboxFactory
    for the 1st parameter of io.druid.indexing.overlord.ThreadPoolTaskRunner.<init>(ThreadPoolTaskRunner.java:100)
  at io.druid.cli.CliPeon$1.configure(CliPeon.java:192) (via modules: com.google.inject.util.Modules$OverrideModule -> com.google.inject.util.Modules$OverrideModule -> io.druid.cli.CliPeon$1)
  while locating io.druid.indexing.overlord.ThreadPoolTaskRunner
  while locating io.druid.indexing.overlord.TaskRunner
    for the 4th parameter of io.druid.indexing.worker.executor.ExecutorLifecycle.<init>(ExecutorLifecycle.java:78)
  at io.druid.cli.CliPeon$1.configure(CliPeon.java:182) (via modules: com.google.inject.util.Modules$OverrideModule -> com.google.inject.util.Modules$OverrideModule -> io.druid.cli.CliPeon$1)
  while locating io.druid.indexing.worker.executor.ExecutorLifecycle
```

In `/opt/druid/conf/druid/_common/common.runtime.properties` there is the `druid.storage.type=local` therefore I would assume that no S3-credentials are needed.

If you really want to support S3 you probably should include `jets3t.properties` and make it configurable. (Or switch to the new 0.13.0 version which uses aws-sdk instead of jet3t.)

**What you expected to happen**:

A simple `helm install incubator/druid` should work out of the box. (And probably simply use local storage instead of S3.)

Using S3 should be possible by specifying all S3 parameters (including an non-AWS endpoint) as values.

**How to reproduce it** (as minimally and precisely as possible):

* Install using `helm install incubator/druid`
* post a proper spec to `druid/indexer/v1/supervisor` 
* watch `/var/druid/indexing-logs` in the middlemanager node

**Anything else we need to know**:

### PR DESCRIPTION
**Pull request by gerbal**
This PR updates the druid chart to use the [druid project docker](https://github.com/apache/incubator-druid/tree/master/distribution/docker). [Built by me](https://cloud.docker.com/u/grantmcl/repository/docker/grantmcl/druid) until the druid project provides their own builds. 

This updated image supports much more flexible configuration via environment variables, including plugin specifications. Most configuration can be freely shared between druid components, so I've moved the environment variables into a configmap. 

I've also added postgresql as a default metadata storage provider

 - fixes #11484 - plugins now set in values

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md

### CODE REVIEW

File: incubator/druid/values.yaml [hunk 1]
```diff
@@ -1,12 +1,33 @@
 # Default values for druid.
 
 image:
-  repository: maver1ckpl/druid-docker
-  tag: 0.12.3-2
+  repository: grantmcl/druid
```
Comment (maver1ck) [inline incubator/druid/values.yaml:4]: We don't want to make this official ?

File: incubator/druid/templates/configmap.yaml [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    release: {{ .Release.Name }}
+    heritage: {{ .Release.Service }}
+data:
+{{ toYaml .Values.configVars | indent 2 }}
+{{- if .Values.zookeeper.enabled }}
+  druid_zk_service_host: {{ .Release.Name }}-zookeeper-headless:2181
+{{- else }}
+  druid_zk_service_host: {{ .Values.zkHosts }}
+{{- end }}
+{{- if .Values.mysql.enabled }}
```
Comment (maver1ck) [inline incubator/druid/templates/configmap.yaml:17]: How to use this Chart with external mysql or postgresql ?

**Review by maver1ck — CHANGES_REQUESTED** [review]

File: incubator/druid/templates/configmap.yaml (hunk 2 above)
Comment (gerbal) [reply incubator/druid/templates/configmap.yaml:17]: all druid configuration variables can be set from the `configVars` section of values.yaml. 

For instance, for an external postgres instance I would set: 

```
configVars:
  druid_metadata_storage_type: postgresql
  druid_metadata_storage_connector_connectURI: jdbc:postgresql://postgres:5432/druid
  druid_metadata_storage_connector_user: druid
  druid_metadata_storage_connector_password: FoolishPassword
```

Any suggestions to make this more explicit?

File: incubator/druid/values.yaml (hunk 1 above)
Comment (gerbal) [reply incubator/druid/values.yaml:4]: I couldn't find [the official image](https://github.com/apache/incubator-druid/tree/master/distribution/docker) in any docker registry, so I built it myself.

I would prefer an official image of course.

**Review by hall — CHANGES_REQUESTED** [review]
@gerbal, will you sign this (latest) commit?

**Comment by remizyaka** [comment]
any news on that thing?

**Comment by ilhanadiyaman** [comment]
Any updates on this PR? It would be much beneficial to use official docker image for helm chart.

**Comment by gerbal** [comment]
I haven't had time to come back to this PR, and I've had some stability issues with my deployment of druid and am not confident endorsing it for others use.  

Plus I would like to wait for an [official released image](https://github.com/apache/incubator-druid/issues/8103) so no one else is dependent on me keeping the underlying image up to date.

**Comment by ilhanadiyaman** [comment]
Official docker image is coming on next release https://github.com/apache/incubator-druid/issues/8103

@gerbal are you planning to update the PR accordingly?

**Pull request closed without merging**